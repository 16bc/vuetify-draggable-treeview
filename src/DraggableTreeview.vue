<template>
  <draggable
    :value="value"
    :group="group"
    class="v-treeview theme-light v-treeview-draggable"
    ghost-class="ghost"
    @input="value => $emit('input', value)"
  >
    <draggable-tree-view-node
      :group="group"
      v-for="item in value"
      :key="item.id"
      :value="item"
      @input="updateItem"
    >
      <template v-slot:prepend="{ item, open }">
        <slot name="prepend" v-bind="{ item, open }"> </slot>
      </template>
      <template v-slot:label="{ item, open }">
        <slot name="label" v-bind="{ item, open }"> </slot>
      </template>
      <template v-slot:append="{ item, open }">
        <slot name="append" v-bind="{ item, open }"> </slot>
      </template>
    </draggable-tree-view-node>
  </draggable>
</template>

<script lang="ts">
import Vue, { PropType } from "vue";
import draggable from "vuedraggable";
import DraggableTreeViewNode from "./DraggableTreeviewNode.vue";

export default Vue.extend({
  components: {
    draggable,
    DraggableTreeViewNode
  },
  props: {
    value: {
      type: Array as PropType<{ id: string | number }[]>,
      default: (): { id: string | number }[] => {
        return [];
      }
    },
    group: {
      type: [String, null],
      default: null
    }
  },
  methods: {
    updateItem(itemValue): void {
      console.log("itemValue");
      console.log(itemValue);
      console.log(itemValue);
      const newValue = this.value.map(v => {
        if (v.id === itemValue.id) {
          return itemValue;
        }
        return v;
      });
      console.log("newValue");
      console.log(newValue);
      this.$emit("input", newValue);
    }
  }
});
</script>
