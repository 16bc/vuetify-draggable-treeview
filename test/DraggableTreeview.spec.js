import Vuetify from "vuetify";
import { mount } from "@vue/test-utils";
import DraggableTreeview from "../src/DraggableTreeview.vue";

describe("VueDraggableTreeview", () => {
  test("renders correctly", () => {
    const wrapper = mount(DraggableTreeview, {
      vuetify: new Vuetify(),
      propsData: {
        value: [
          {
            id: 1,
            name: "test",
            children: [{ id: 101, name: "test-children" }]
          }
        ]
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
