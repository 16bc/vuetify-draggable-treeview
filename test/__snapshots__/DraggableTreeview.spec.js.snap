// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyField renders correctly 1`] = `
<div
  class="v-treeview theme-light v-treeview-draggable"
>
  <div
    class="v-treeview-node v-treeview-node--click "
  >
    <div
      class="v-treeview-node__root"
    >
      <i
        class="v-icon notranslate v-treeview-node__toggle v-icon--link mdi mdi-menu-down theme--light"
        role="button"
      />
        
      <div
        class="v-treeview-node__label"
      >
        
        test
      
      </div>
       
    </div>
     
    <!---->
  </div>
</div>
`;
